kombu/transport/azureservicebus.py

```diff
@@ -67,7 +67,8 @@
 
 # dots are replaced by dash, all other punctuation replaced by underscore.
 CHARS_REPLACE_TABLE = {
-    ord(c): 0x5f for c in string.punctuation if c not in '_'
+    ord('.'): ord('-'),
+    **{ord(c): ord('_') for c in string.punctuation if c not in ['_', '.', '-']}
```

Comment:
We can precalculate the punctuations we need to replace.
This will make the module's load time slightly faster and it is also cleaner.

```python
PUNCTUATIONS_TO_REPLACE = set(string.punctuation) - {'_', '.', '-'}

CHARS_REPLACE_TABLE = {
  ord('.'): ord('-'),
  **{ord(c): ord('_') for c in PUNCTUATIONS_TO_REPLACE}
}
```

Reply:
This looks good, should work great.

Reply:
Thanks for suggestion, I'll change it accordingly. And will add tests 😀

Reply:
@thedrow, I've changed the flow to your suggestions, I also added tests :)